Tiltfile

```diff
@@ -169,125 +214,203 @@ WORKDIR /
-            port_forwards = port_forwards,
-            links = links,
-            resource_deps = ["provider_crd"],
-        )
+	deployment_kind = "provider"
+	p = providers.get(name)
+	label = p.get("label")
+
+	port_forwards, links = get_port_forwards(debug)
+
+	build_go_binary(
+		name = label,
+		context = p.get("context"),
+		reload_deps = p.get("live_reload_deps"),
+		debug = debug,
+		go_main = p.get("go_main", "main.go"),
+		binary_name="manager",
+		label = label
+	)
+
+	build_docker_image(
+		image = p.get("image"),
+		context = p.get("context"),
+		binary_name = "manager",
+		additional_docker_helper_commands = p.get("additional_docker_helper_commands", ""),
+    	additional_docker_build_commands = p.get("additional_docker_build_commands", ""),
+		port_forwards = port_forwards
+	)
+
+	if p.get("kustomize_config", True):
+		yaml = read_file("./.tiltbuild/yaml/{}.{}.yaml".format(name, deployment_kind))
+		k8s_yaml(yaml)
+		objs = decode_yaml_stream(yaml)
+		k8s_resource(
+			workload = find_object_name(objs, "Deployment"),
+			objects = [find_object_qualified_name(objs, "Provider")],
+			new_name = label.lower() + "_controller",
+			labels = [label, "ALL.controllers"],
+			port_forwards = port_forwards,
+			links = links,
+			resource_deps = ["provider_crd"],
+		)
+
+def enable_addons():
+    for name in settings.get("deploy_addons", []):
+		addon = addons.get(name)
+		debug = addon.get("debug", {})
+		print(debug)
+		deployment_kind = "addon"
+		label = addon.get("label")
+		port_forwards, links = get_port_forwards(debug)
+
+		build_go_binary(
+			name = name,
+			context = addon.get("context"),
+			reload_deps = addon.get("live_reload_deps"),
+			debug = debug,
+			go_main = addon.get("go_main", "main.go"),
+			binary_name=addon.get("binary_name"),
+			label = label
+		)
+
+		build_docker_image(
+			image = addon.get("image"),
+			context = addon.get("context"),
+			binary_name = addon.get("binary_name"),
+			additional_docker_helper_commands = addon.get("additional_docker_helper_commands", ""),
+			additional_docker_build_commands = addon.get("additional_docker_build_commands", ""),
+			port_forwards = port_forwards
+		)
+
+		if addon.get("kustomize_config", True):
+			yaml = read_file("./.tiltbuild/yaml/{}.{}.yaml".format(name, deployment_kind))
+			k8s_yaml(yaml)
+			objs = decode_yaml_stream(yaml)
+			k8s_resource(
+				workload = find_object_name(objs, "Deployment"),
+				new_name = name.lower() + "_addon",
+				labels = [label, "ALL.addons"],
+				port_forwards = port_forwards,
+				links = links
+				)
+
+		additional_resources = addon.get("additional_resources", [])
+		for resource in additional_resources:
+			yaml = read_file(addon.get("context") + "/" + additional_resources[resource])
+			k8s_yaml(yaml)
+			objs = decode_yaml_stream(yaml)
+			k8s_resource(
+				workload = name.lower() + "_addon",
+				objects = [find_object_qualified_name(objs, resource)],
+				resource_deps = ["capi_controller"],
```

Comment:
This hard-coded dependency doesn't feel very generic (it's because of the ExtensionConfig CRD, right?)